python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -415,7 +415,23 @@ def mult(df):
         chunks = [
             grouped_values[s:e] for s, e in zip(offsets[:-1], offsets[1:])
         ]
-        result = cudf.concat([function(chk) for chk in chunks])
+        chunk_results = [function(chk) for chk in chunks]
+
+        if len(chunk_results) > 0 and cudf.utils.dtypes.is_scalar(
+            chunk_results[0]
+        ):
+            result = cudf.Series(
+                chunk_results, index=self.grouping.keys.unique()
```

Comment:
Does using `self.grouping.keys.unique()` guarantee that the keys are ordered the same as `chunk_results` here? This feels like an implementation detail that we're reliant on.

Reply:
My bad, yes `unique` seems to call `drop_duplicates` which is not guaranteed to preserve order. Should we build an internal utility something like:
```
   def _ordered_unique(array):
	df = cudf.DataFrame({'a':array, 'b':column.arange(len(array))})
	df = df.drop_duplicates('a')
	df = df.sort_values(by=['b'])
	return df['a']
```

Reply:
I want to say that `self.grouping.keys` should already be sorted in that we should be able to use the offsets to just grab the keys as needed.

Reply:
Understood, made the change to extract from offsets